### ISSUE TITLE
Fritzbox Tools (latest) broke RESTful switches / sensors (?)

### ISSUE THREAD

**Issue by agrimpelhuber** [mammuth/ha-fritzbox-tools#103]
On my HassOS-based RPi 3B installation of HA, one of the recent updates broke all RESTful switches and sensors. Even though CURL-based calls worked from within the HA container, the setup of those entities ended up in (please note: No DNS involved):

```
2020-06-25 08:02:59 ERROR (MainThread) [homeassistant.components.rest.switch] No route to resource/endpoint: http://192.168.178.168/switch.json
```

The async / aiohttp messages are not very specific at this point, but the most likely cause was a timeout of the relevant HTTP calls. Extending the timeout to more than 5 minutes didn't help, though. It's also remarkable that all switches did not work at all - not any of them.

After finding no solution, I ended up re-installing HA piece by piece on a new machine (RPi 3B+). Everything worked fine for a long time, until I installed HACS and the previously installed HACS-based integrations. That's when the RESTful communication broke again.

Removing and re-installing the integrations one by one, I ended up here: Whenever I re-install Fritzbox Tools, the RESTful switches stop working. No idea why. Here's one of my switch configurations. I know that it's a bit complicated, taking URL-encoded POST data and returning JSON, but it worked. However, even the most simple switch will be affected the same way.

```
switch:
  - platform: rest
    name: heating_override
    timeout: 20
    resource: 'http://192.168.178.202/switch.json'
    method: post
    body_on: 'state=on'
    body_off: 'state=off'
    is_on_template: '{{ value_json.logicstate }}'
    headers:
      Accept: text/json
      Content-Type: application/x-www-form-urlencoded
```

**Comment by mammuth** [comment]
Wow, what a strange issue. Sorry for the extensive troubleshooting you had to go through.

I don't see any obvious reason for this behavior :thinking:

Does setting the log level of `homeassistant.components.rest.switch` and `custom_components.fritzbox_tools` to `debug` lead to any additional insights?

**Comment by agrimpelhuber** [comment]
Setting `homeassistant.components.rest.switch` to `debug` just brought the insight that it might be a timeout issue. The fact that the RESTful switch reports `No route to resource/endpoint` is in fact a fake - only the DEBUG setting actually revealed the timeout.

I didn't try the DEBUG setting with Fritzbox Tools after I had discovered the obvious problem with them - simply because I'm mostly interested in having my installation running. I could test it with the old, failing installation, but it's in conflict with a few devices that can only talk to one HA installation.

Fritzbox Tools is starting just fine, and reports all the relevant data and switches.

I see that this does not help a lot to debug, so I set up a Hyper-V HA instance and tested a backup of the old installation. It doesn't work properly, and the most I can make out of the logs is that there are a lot of failed asynchronous HTTP calls. The most I can make of it is that there is just a limited number of calls available, and a lot of slow ones block others from finishing.

The relation between RESTful switches and Fritz Tools seems just coincidental (maybe because my FritzBox is very slow), so I'm closing this issue.

Thanks!
